src/librustdoc/clean/mod.rs

```diff
@@ -1868,13 +1871,41 @@ impl Clean<Item> for doctree::Trait {
                 items: self.items.clean(cx),
                 generics: self.generics.clean(cx),
                 bounds: self.bounds.clean(cx),
-                is_spotlight: is_spotlight,
+                is_spotlight,
                 is_auto: self.is_auto.clean(cx),
             }),
         }
     }
 }
 
+#[derive(Clone, RustcEncodable, RustcDecodable, Debug)]
+pub struct TraitAlias {
+    pub generics: Generics,
+    pub bounds: Vec<GenericBound>,
+    pub is_spotlight: bool,
+}
+
+impl Clean<Item> for doctree::TraitAlias {
+    fn clean(&self, cx: &DocContext) -> Item {
+        let attrs = self.attrs.clean(cx);
+        let is_spotlight = attrs.has_doc_flag("spotlight");
```

Comment:
We should check if spotlight actually works for trait impls using an aliased trait.

So, we should see what happens for:

```rust
#[doc(spotlight)]
trait Foo {}
trait Bar = Foo;
struct Ty;
impl Bar for Ty;
```

and 

```rust
trait Foo {}
#[doc(spotlight)]
trait Bar = Foo;
struct Ty;
impl Bar for Ty;
```

Reply:
You can't implement a trait alias directly. So I guess that spotlight doesn't make sense for a trait alias. I'll remove spotlight "support" for them.